clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -198,6 +215,20 @@ protected void handleFetchSuccess(final Node fetchTarget,
                 fetchBuffer.add(completedFetch);
             }
 
+            if (!partitionsWithUpdatedLeaderInfo.isEmpty()) {
```

Comment:
I'm a bit confused why we collect the `partitionsWithUpdatedLeaderInfo` above when it looks like all we do with them is validate them to the subscriptions later. Is there any other use for having it out of the loop?

Reply:
`partitionsWithUpdatedLeaderInfo` are collected above to update metadata once with all updates via `metadata.updatePartitionLeadership`.